pkg/clusteragent/admission/mutate/autoinstrumentation/language_versions.go

```diff
@@ -48,6 +51,23 @@ func (l language) libInfo(ctrName, image string) libInfo {
 	}
 }
 
+// DEV: Will attempt to resolve, defaults to legacy if unable
+func (l language) libInfoWithResolver(ctrName, registry string, version string, imageResolver ImageResolver) libInfo {
+	resolvedImage, ok := imageResolver.Resolve(registry, fmt.Sprintf("dd-lib-%s-init", l), version)
+	var image string
+	if !ok {
+		log.Warnf("failed to resolve image %s/%s:%s, using fallback", registry, fmt.Sprintf("dd-lib-%s-init", l), version)
```

Comment:
In the case there is a noop resolver, this will create a lot of warnings. We likely will need a way to only log if it's supposed to be using RC and is not able to.